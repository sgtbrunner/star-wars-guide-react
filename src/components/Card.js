import React from 'react';
import { getImage } from '../utils/functions.utils';

const Card = ({ name, index, openModal }) => (
  <div onClick={openModal}>
    <div className="card" id={index}>
      <img className="picture" src={getImage(index)} alt="character-portrait" id={index} />
      <div className="title" id={index}>
        {name}
      </div>
    </div>
  </div>
);

export default Card;
