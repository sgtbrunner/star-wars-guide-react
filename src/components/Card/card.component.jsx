import React from 'react';
import PropTypes from 'prop-types';

import { getImageUrl } from '../../utils/functions.utils';
import './card.styles.css';

const Card = ({ name, index, openDialog }) => (
  <button type="button" onClick={openDialog} className="card pointer" id={index}>
    <img src={getImageUrl(index)} alt="character-portrait" id={index} />
    <div className="name" id={index}>
      {name}
    </div>
  </button>
);

Card.propTypes = {
  name: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  openDialog: PropTypes.func.isRequired,
};

export default Card;
