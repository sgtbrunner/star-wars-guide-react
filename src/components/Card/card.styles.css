.card {
  width: 210px;
  height: 100%;
  border: 3px solid goldenrod;
  margin: 15px 5px;
  padding: 0;
  display: flex;
  flex-direction: column;
  user-select: none;
}

.card:hover {
  transform: scale(1.05);
  box-shadow: rgba(255, 255, 255, 2) 0px 5px 15px, inset rgba(255, 255, 255, 2) 0px -10px 20px;
}

.name {
  background-color: white;
  padding: 5px;
  width: 100%;
  font-family: Verdana;
  font-size: 15px;
  font-weight: bold;
}
