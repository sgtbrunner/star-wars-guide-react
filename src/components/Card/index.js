import { connect } from 'react-redux';

import { openDialog } from '../../redux/dialog/dialog.actions';
import Card from './card.component';

const mapDispatchToProps = (dispatch) => ({
  openDialog: (card) => openDialog(dispatch, Number(card.target.id)),
});

export default connect(null, mapDispatchToProps)(Card);
