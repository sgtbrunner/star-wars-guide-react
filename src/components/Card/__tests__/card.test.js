import React from 'react';
import { render, screen } from '@testing-library/react';
import Card from '../card.component';

const props = {
  name: 'Teste',
  index: 0,
  openDialog: jest.fn(),
};

const mountCardComponent = (newProps) => render(<Card {...props} {...newProps} />);

describe('Card component', () => {
  it('should match snapshot', () => {
    const wrapper = mountCardComponent();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render component without errors', () => {
    mountCardComponent();

    const card = screen.getByRole('button');
    const cardImage = screen.getByAltText('character-portrait');
    const cardTitle = screen.getByText(props.name);

    expect(card).toBeInTheDocument();
    expect(cardImage).toBeInTheDocument();
    expect(cardTitle).toBeInTheDocument();
  });
});
