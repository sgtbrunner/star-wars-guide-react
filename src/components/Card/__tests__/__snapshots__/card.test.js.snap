// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="card pointer"
        id="0"
        type="button"
      >
        <img
          alt="character-portrait"
          id="0"
          src="https://starwars-visualguide.com/assets/img/characters/1.jpg"
        />
        <div
          class="name"
          id="0"
        >
          Teste
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="card pointer"
      id="0"
      type="button"
    >
      <img
        alt="character-portrait"
        id="0"
        src="https://starwars-visualguide.com/assets/img/characters/1.jpg"
      />
      <div
        class="name"
        id="0"
      >
        Teste
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
